src/mame/apple/apple2e.cpp

```diff
@@ -1602,37 +1605,42 @@ void apple2e_state::lc_update(int offset, bool writing)
 
 	if (m_lcram != old_lcram)
 	{
-		if (m_iscec)
+		lcrom_update();
+	}
+
+	#if 0
+	printf("LC: new state %c%c dxxx=%04x altzp=%d (PC=%x)\n",
+			m_lcram ? 'R' : 'x',
+			m_lcwriteenable ? 'W' : 'x',
+			m_lcram2 ? 0x1000 : 0x0000,
+			m_altzp, m_maincpu->pc());
+	#endif
```

Comment:
Please avoid `printf` – it isn’t type-safe.  Also try to avoid using the preprocessor to disable logging code like this as it will rot when it isn’t enabled.  The utilities from `"logmacro.h"` should be used for compile-time configurable logging.

Reply:
Not my code but good to know. I'll remove the logging in another PR coming in soon.

Reply:
Modernizing the logging definitely should be a separate PR.  You can see how the modern logging with the different channels works in e.g. apple/dafb.cpp.  I'll check the rest of this PR in a moment.